app.rb

```diff
@@ -0,0 +1,150 @@
+require 'date'
+require_relative 'person_manager'
+require_relative 'book_manager'
+require_relative 'rental_manager'
+
+# app.rb
+class App
+  def initialize
+    @person_manager = PersonManager.new
+    @book_manager = BookManager.new
+    @rental_manager = RentalManager.new
+  end
+
+  def exit_app
+    puts 'Exiting the Library App. Goodbye!'
+    exit
+  end
+
+  def run
+    display_welcome_message
+
+    loop do
+      print_options_menu
+      option = gets.chomp.to_i
+      process_option(option)
+      break if exit_option?(option)
+    end
+
+    display_goodbye_message
+  end
+
+  private
+
+  def display_welcome_message
+    puts ''
+    puts 'Welcome to the Library App!'
+  end
+
+  def print_options_menu
+    puts ''
+    print 'Please choose an option by selecting a number: '
+    puts "\nOptions:"
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person'
+    puts '4. Create a book'
+    puts '5. Create a rental'
+    puts '6. List rentals for a person with a given id'
+    puts '7. Exit'
+  end
+
+  def exit_option?(option)
+    option == 7
+  end
```

Comment:
- [ ] As suggested by the previous reviewer, please kindly make sure that the part of your code where you present the users with a list of options to perform is written in the main.rb file.

Reply:
Hi @topeogunleye , Thank you for your review. While, I am trying to make these change, I am not sure what you mean when you say "make sure that the part of your code where you present the users with a list of options to perform is written in the main.rb file. This is a good practice, as it makes your code more modular and readable." I am a bit worried if this matches any of the project requirement. That part of the project requirement about main goes as follows: In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
Present the user with a list of options to perform.
Lets users choose an option.
If needed, ask for parameters for the option.
Have a way to quit the app.
A method, not a class or anything else. NOT A FILE TO CARRY A PARTCULAR CODE. A METHOD THAT WHEN EVOKED (CALLED at the end of the file) WILL PRESENT THE USER WITH A MENU OPTIONS..  So I am not sure what you mean here

Reply:
PLease I will need you to clarify this.

Reply:
@enocol, Hi, Thank you very much, as you have stated, In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
`Present the user with a list of options to perform.` 

Keyword: Present the user with a list of options to perform, and your functionality for presenting the user with a list of list of options to perform is in the app.rb file instead of the main.rb

Reply:
Hi @topeogunleye  Can you clarify this recommendation please. I would suggest we have a zoom or or in chat for 5 minutes to see if what you recommeded is part of the project requirement. Because right now, you are asking me to put some code in the main.rb file. The main.rb file as far as I understand is suppose to be any entry point, and its a method that when invoked, should present the user with those options to choose from, not that it should carry a particular code.

Reply:
> Hi @topeogunleye Can you clarify this recommendation please. I would suggest we have a zoom or or in chat for 5 minutes to see if what you recommeded is part of the project requirement. Because right now, you are asking me to put some code in the main.rb file. The main.rb file as far as I understand is suppose to be any entry point, and its a method that when invoked, should present the user with those options to choose from, not that it should carry a particular code.

Yes, It is, because you have to put the code to `*Lets users choose an option.` in the `main.rb file`, and not the `app.rb`

Reply:
@topeogunleye - PRESENT THE USER WITH A LIST OF OPTIONS TO PERFORM (not put a code ) Where yo u have highlightedi= in red says as follows:
The main.rb file should be an entry point
Its a method that when invoked, should present the user with options to choose etc

Reply:
Hi Mr Enoh,

I hope this message finds you well. Following up on the recent discussions, it's important to clarify that the project requirements emphasize having all the code for starting the app, including the menu items, in the main.rb file and not in the app.rb file. This means that the user interface and menu options should be implemented in the main.rb file as an entry point to your console app.

Please ensure that your main.rb file contains the necessary code for presenting users with menu items and options as specified in the project requirements. If you have any further questions or need clarification, feel free to reach out.

Best regards,
Temitope